### ISSUE TITLE
Object must include `data` key

### ISSUE THREAD

**Issue by ming-codes** [marshmallow-code/webargs#422]
```
{'errors': [{'detail': 'Object must include `data` key.', 'source': {'pointer': '/'}}]}
```

Hi,

I'm using `marshmallow-jsonapi` with [`flask-rest-api`](https://github.com/Nobatek/flask-rest-api). I encountered the above error when trying to load a JSONAPI document from location json.

I'm not sure how `webargs` is supposed to work. It looks like it parses data twice. [once from each field](https://github.com/marshmallow-code/webargs/blob/5.4.0/src/webargs/core.py#L283) and [once again from schema.load](https://github.com/marshmallow-code/webargs/blob/5.4.0/src/webargs/core.py#L369) using the same schema. This could work for schemas where field and input matches one to one, but it's not the case JSON API.

What would be to proper way to make this work?

**Comment by sirosen** [comment]
I'm not familiar with `marshmallow-jsonapi`, but it looks like it uses `pre_load` hooks. One of the major changes in webargs v6 is that `pre_load` and `post_load` hooks are invoked correctly.

If you try the webargs v6 beta -- version `6.0.0b6` right now -- you should see things working. But definitely post again here if it isn't, as there's still (hopefully only a little bit of) time before v6 drops and we can make changes more easily while it's still in beta.

**Comment by lafrech** [comment]
Assuming this is fixed in 6.0.0 beta. Please comment if it is still an issue and we'll reopen.
